Gems/Atom/RPI/Code/Source/RPI.Public/Pass/PassSystem.cpp

```diff
@@ -323,32 +251,37 @@ namespace AZ
 
             m_state = PassSystemState::FrameEnd;
 
-            m_rootPass->FrameEnd();
+            for (RenderPipeline*& pipeline : m_renderPipelines)
+            {
+                pipeline->PassSystemFrameEnd();
+            }
+            m_passesWithoutPipeline.m_rootPass->FrameEnd();
 
             // remove any pipelines that are marked as ExecuteOnce
-            // the immediate children of m_rootPass are the top-level nodes of each pipeline
-            for (const Ptr<Pass>& pass : m_rootPass->GetChildren())
+            for (RenderPipeline*& pipeline : m_renderPipelines)
             {
-                RenderPipeline* pipeline = pass->m_pipeline;
                 if (pipeline && pipeline->IsExecuteOnce())
                 {
                     pipeline->RemoveFromScene();
                 }
             }
 
-            m_passHierarchyChanged = false;
-
             m_state = PassSystemState::Idle;
         }
 
+        // --- Misc --- 
+
         void PassSystem::Shutdown()
         {
-            RemovePasses();
-            m_buildPassList.clear();
+            m_passesWithoutPipeline.ClearQueues();
+            m_passesWithoutPipeline.m_rootPass = nullptr;
             m_rootPass = nullptr;
+
+            AZ_Assert(m_passCounter == 0, "Pass leaking has occurred! There are %d passes that have not been deleted.\n", m_passCounter);
+
```

Comment:
Is anything needed to clear out the render pipeline references and their passes?

Reply:
Pipeline's all shut down (and thus remove themselves) before the pass system shuts down. Can add an assert to ensure this stays the case.